cmd/internal/find/primary/meta/objectPredicate.go

```diff
@@ -71,6 +53,51 @@ func parseObjectP(tokens []string, baseCaseParser, keySequenceParser predicate.P
 	return objectP(key, p), tokens, err
 }
 
+// A key consists of one or more characters that aren't
+// ".", "[", or "]". The reason we have this limitation
+// is so we can support key sequences, which includes
+// nested keys like ".key1.key2", ".key1[?].key2", etc.
+// key sequences aren't specified in the grammar because
+// they make it harder to formalize the semantics.
+//
+// NOTE: Users can still specify ".", "[", or "]" by escaping
+// them with a backslash "\". For example, '.com\.docker\.compose'
+// would be parsed as the key "com.docker.compose".
+func parseKeySequence(tk string) (string, string, error) {
```

Comment:
The name could reflect that it's only parsing part of the string, like `nextKeyFromSequence`.

Reply:
Renamed to `parseKey` -- I think the return type should make it clear that it's parsing part of the string to get the key.